fusion-plugin-react-router/src/modules/BrowserRouter.js

```diff
@@ -58,6 +48,7 @@ class BrowserRouter extends React.Component<PropsType> {
     const {Provider, history, basename} = this.props;
     if (!Provider) throw new Error('Missing Provider for Browser Router');
     return (
+      // $FlowFixMe
```

Comment:
Necessary to resolve the following Flow complaint:

```sh
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/modules/BrowserRouter.js:51:8

Cannot create Provider element because property basename is missing in object type [1] but exists in props [2].

     src/modules/BrowserRouter.js
      48│     const {Provider, history, basename} = this.props;
      49│     if (!Provider) throw new Error('Missing Provider for Browser Router');
      50│     return (
 [2]  51│       <Provider basename={basename} history={history}>
      52│         {this.props.children}
      53│       </Provider>
      54│     );
      55│   }
      56│ }

     src/types.js
 [1] 148│ type BaseRouterType = React.ComponentType<{|
     149│   history: RouterHistoryType,
     150│   children?: React.Node,
     151│ |}>;
```

Note that`BaseRouterType` is taken directly from the libdef on flow-typed:

https://github.com/flow-typed/flow-typed/blob/c65b1c9cdf830e095ebc359130f1d3330428cf51/definitions/npm/react-router-dom_v4.x.x/flow_v0.98.x-/react-router-dom_v4.x.x.js#L121-L124

It isn't clear to me if `basename` should be added.  Happy to hear others' thoughts here.

Reply:
This is definitely confusing. It looks like the base `Router` takes history and children props (https://reacttraining.com/react-router/web/api/Router), but if the user was to supply their own router: https://github.com/fusionjs/fusionjs/blob/d25d2c6ada528d153ae1bda41a16be2c9bbce885/fusion-plugin-react-router/src/plugin.js#L55 then basename should be supplied because it could be one of the other router types (`BrowserRouter`, `HashRouter`, etc.) that do take `basename` prop